packages/history/tests/createHashHistory.test.ts

```diff
@@ -0,0 +1,32 @@
+import { describe, expect, test } from 'vitest'
+
+import { createHashHistory } from '../src'
+
+describe('createHashHistory', () => {
+    describe('parseLocation', () => {
+        describe.each([
+            ['/', {pathname: '/', search: ''}, 'neither search params nor hash'],
+            ['/#hello', {pathname: '/hello', search: ''}, 'hash present, no search params'],
+            ['/?search=params', {pathname: '/', search: '?search=params'}, 'search params present, no hash'],
+            ['/#hello?search=params', {pathname: '/hello', search: '?search=params'}, 'both hash and search params present, in that order'],
+            ['/?search=params#hello', {pathname: '/hello', search: '?search=params'}, 'both search params and hash present, in that order'],
+        ])('check for %s', (...[path, exp, desc]) => {
+            test(`onLoad with ${path} (${desc})`, () => {
+                const mockWindow = {
+                    addEventListener: window.addEventListener,
+                    history: window.history,
+                    location: new URL(`https://www.example.com${path}`)
```

Comment:
@SeanCassiere The search params update triggers the page refresh, but such functionality isn't implemented in JSDOM, and it just throws an error. Thus, it requires mocking over the location.

I found [this stackover flow post](https://stackoverflow.com/questions/54021037/how-to-mock-window-location-href-with-jest-vuejs) and took its idea to use the URL API to mock the location API. Please let me know if you have some concerns or other better ideas for it. Thanks.

Reply:
It might just be missing the correct setup for the testing env.

Could you refactor to using `window.location` and then ping me? I'll fix up the deps and get the testing environment working.

Reply:
So the JSDOM didn't implement the navigation functionality by design (see [here](https://github.com/jsdom/jsdom/issues/2112)).

Basically, the `window` object is an object mocked by JSDOM because it exists only in the browser environment. Are you saying that the such mocking exists under the `packages/react-router`? For all test cases under that package, it uses `history` object instead of window.location. Could you point me to the code block if I missed it?

Reply:
You are right, my mistake. I was thinking about this https://github.com/TanStack/router/blob/35c7c495f38f89675a5570a9b4fa35e63f0a3525/packages/react-router/tests/link.test.tsx#L51

In that case, don't mock `window`, just do a `window.(push/replace)State` before calling `createHashHistory` for the "on-load" tests.

Reply:
Alright, I've updated the PR. Please take a look when you get some time. Thanks.